From 4d2b36054d9c80c7a29b1afcdf842ae46112bbb0 Mon Sep 17 00:00:00 2001
From: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Date: Wed, 5 Jul 2017 16:31:42 +0300
Subject: [PATCH 2234/5242] MLK-15986-4: video: fbdev: Separate display
 framework code from fb driver code

commit  72e6e73cb614dc1fb4fb37c809863f926b542448 from
https://source.codeaurora.org/external/imx/linux-imx.git

The display driver framework code in drivers/video/fbdev/mxc/mxc_dispdrv.c
is currently compiled along with fb driver code in
drivers/video/fbdev/mxc. This is a framework that is also used by another
fb driver that can be found in drivers/video/fbdev/mxsfb.c.

This patch separates the framework compilation by pulling it into its
own CONFIG_FB_MXC_DISP_FRAMEWORK that can be selected by any fb driver
and/or panel whose implementation relies on this framework.

Signed-off-by: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/Kconfig      |    2 +-
 drivers/video/fbdev/mxc/Kconfig  |   10 ++++++++++
 drivers/video/fbdev/mxc/Makefile |    3 ++-
 3 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/Kconfig b/drivers/video/fbdev/Kconfig
index 4e10e03..c72c94d 100644
--- a/drivers/video/fbdev/Kconfig
+++ b/drivers/video/fbdev/Kconfig
@@ -2264,7 +2264,7 @@ config FB_JZ4740
 config FB_MXS
 	tristate "MXS LCD framebuffer support"
 	depends on FB && (ARCH_MXS || ARCH_MXC)
-	depends on FB_MXC_SYNC_PANEL
+	select FB_MXC_DISP_FRAMEWORK
 	select FB_CFB_FILLRECT
 	select FB_CFB_COPYAREA
 	select FB_CFB_IMAGEBLIT
diff --git a/drivers/video/fbdev/mxc/Kconfig b/drivers/video/fbdev/mxc/Kconfig
index 6f6589e..33ff7db 100644
--- a/drivers/video/fbdev/mxc/Kconfig
+++ b/drivers/video/fbdev/mxc/Kconfig
@@ -14,9 +14,19 @@ config FB_MXC
 	  If you plan to use the LCD display with your MXC system, say
 	  Y here.
 
+config FB_MXC_DISP_FRAMEWORK
+	tristate "Display driver framework"
+	help
+	  This is a framework that helps with registration and data handling
+	  between fb drivers and display drivers.
+
+	  It is selected by drivers which use this framework as linkage code
+	  between display controllers and panels.
+
 config FB_MXC_SYNC_PANEL
 	depends on FB_MXC
 	depends on MXC_IPU_V3
+	select FB_MXC_DISP_FRAMEWORK
 	tristate "Synchronous Panel Framebuffer"
 
 config FB_MXC_OVERLAY
diff --git a/drivers/video/fbdev/mxc/Makefile b/drivers/video/fbdev/mxc/Makefile
index a910a3a..a36b323 100644
--- a/drivers/video/fbdev/mxc/Makefile
+++ b/drivers/video/fbdev/mxc/Makefile
@@ -8,7 +8,8 @@ obj-$(CONFIG_FB_MXC_LDB) += ldb.o
 obj-$(CONFIG_FB_MXC_HDMI)			+= mxc_hdmi.o
 obj-$(CONFIG_FB_MXC_EDID)			+= mxc_edid.o
 obj-$(CONFIG_FB_MXC_ADV7535)			+= adv7535.o
-obj-$(CONFIG_FB_MXC_SYNC_PANEL) += mxc_dispdrv.o mxc_lcdif.o mxc_ipuv3_fb.o
+obj-$(CONFIG_FB_MXC_DISP_FRAMEWORK)		+= mxc_dispdrv.o
+obj-$(CONFIG_FB_MXC_SYNC_PANEL)			+= mxc_lcdif.o mxc_ipuv3_fb.o
 obj-$(CONFIG_FB_MXC_EINK_PANEL)			+= mxc_epdc_fb.o
 obj-$(CONFIG_FB_MXC_EINK_V2_PANEL)		+= mxc_epdc_v2_fb.o
 obj-$(CONFIG_FB_MXS_SII902X) += mxsfb_sii902x.o
-- 
1.7.9.5

